ag/enrollment.go

```diff
@@ -86,3 +86,25 @@ func (m *Enrollment) IsTeacher() bool {
 func (m *Enrollment) IsStudent() bool {
 	return m.GetStatus() == Enrollment_STUDENT
 }
+
+// GetCourseID returns the course ID for a slice of enrollments
+func (m *Enrollments) GetCourseID() uint64 {
+	if len(m.GetEnrollments()) == 0 {
+		return 0
+	}
+	return m.Enrollments[0].GetCourseID()
+}
+
+// HasCourseID checks all enrollments have the same Course ID
+func (m *Enrollments) HasCourseID() bool {
+	if len(m.GetEnrollments()) == 0 {
```

Comment:
I like to be on the safe side and call `m.GetEnrollments()` only once instead of three times (I'm not sure if the compiler is always able to optimize these calls...). Same applies to `GetCourseID()` above.

That is: 
```go
enrollments := m.GetEnrollments()
if len(enrollments) == 0 {
  return false
}
courseID := enrollments[0].GetCourseID()
for _, e := enrollments {
  ...
```